app/__test__/product/product.schema.test.ts

```
@@ -0,0 +1,253 @@
+});
+
+it('should validate a valid input for updateProductSchema', () => {
+  const validInput = {
+    description: 'Updated Product Description',
+    ProductCategory: [
+      {
+        category_id: 2
+      }
+    ]
+  };
+
+  const result = updateProductSchema.safeParse(validInput);
+
+  expect(result.success).toBe(true);
+});
+
+it('should validate a valid input for filterQuery', () => {
+  const validInput = {
+    limit: 5,
+    page: 2
+  };
+
+  const result = filterQuery.safeParse(validInput);
+
+  expect(result.success).toBe(true);
+});
+
+it('should throw an error when the product name has length greater than 50', () => {
+  const invalidInput = {
+    name: 'Product Name with a very long length that exceeds the maximum limit of 50 characters',
+    description: 'Product Description',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Maximum characters are 50');
+});
+
+it('should throw an error when the product description has length less than 20', () => {
+  const invalidInput = {
+    name: 'Product Name',
+    description: 'Product',
+    price: 10,
+    ProductCategory: [
+      {
+        category_id: 1
+      }
+    ],
+    Attribute: [
+      {
+        size: 'M',
+        color: '#FFFFFF',
+        quantity: 5,
+        image: 'https://example.com/image.jpg'
+      }
+    ]
+  };
+
+  const result = createProductSchema.safeParse(invalidInput);
+
+  expect(result.success).toBe(false);
+  expect(result.error.issues[0].message).toBe('Minium of 20 characters required');
+});
+
+it('should throw an error when the product description has length greater than 200', () => {
```

Improve the  unit test description, Ex:
`it('should throw an error when the product description exceeds the maximum length of 200 characters', () => {
`